components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Extensions/MonitoredItemNotificationModelEx.cs

```diff
@@ -35,10 +35,6 @@ public static class MonitoredItemNotificationModelEx {
                 if (message == null) {
                     continue;
                 }
-                if (message.DiagnosticInfo == null && notification.DiagnosticInfos != null &&
```

Comment:
Why did you remove this, was it causing issues or is it never null?   We want to flow the diagnostics info back to the configuration eventually so having this does not hurt on the way there.

Reply:
The diagnostics (if any) are already added to the message through: public static MonitoredItemNotificationModel ToMonitoredItemNotification(). Diagnostics are copied from the DataChangeNotification.DiagnosticInfo array to the MonitoredItemNotification.DiagnosticInfo by the stack before the callback is triggered.